tests/manager/models/test_etcd.py

```diff
@@ -0,0 +1,233 @@
+            }
+        }
+    """.replace("$CONTAINER_REGISTRY_FIELDS", CONTAINER_REGISTRY_FIELDS)
+
+    variables = {
+        "hostname": "cr.example.com",
+        "props": {
+            "username": "username2",
+        },
+    }
+
+    response = await client.execute_async(query, variables=variables, context_value=context)
+    container_registry = response["data"]["modify_container_registry"]["container_registry"]
+    assert container_registry["hostname"] == "cr.example.com"
+    assert container_registry["config"]["url"] == "http://cr.example.com"
+    assert container_registry["config"]["type"] == "dockerhub"
+    assert container_registry["config"]["project"] == ["default"]
+    assert container_registry["config"]["username"] == "username2"
+    assert container_registry["config"]["ssl_verify"] is False
+
+    variables = {
+        "hostname": "cr.example.com",
+        "props": {
+            "type": "harbor2",
+            "project": ["default", "example"],
+        },
+    }
+
+    response = await client.execute_async(query, variables=variables, context_value=context)
+    container_registry = response["data"]["modify_container_registry"]["container_registry"]
+    assert container_registry["hostname"] == "cr.example.com"
+    assert container_registry["config"]["url"] == "http://cr.example.com"
+    assert container_registry["config"]["type"] == "harbor2"
+    assert container_registry["config"]["project"] == ["default", "example"]
+    assert container_registry["config"]["username"] == "username2"
+    assert container_registry["config"]["ssl_verify"] is False
+
+
+@pytest.mark.dependency(depends=["test_modify_container_registry"])
+@pytest.mark.asyncio
+async def test_modify_container_registry_allows_empty_string(
+    client: Client, context: GraphQueryContext
+):
+    query = """
+        mutation ModifyContainerRegistry($hostname: String!, $props: ModifyContainerRegistryInput!) {
+            modify_container_registry(hostname: $hostname, props: $props) {
+                $CONTAINER_REGISTRY_FIELDS
+            }
+        }
+    """.replace("$CONTAINER_REGISTRY_FIELDS", CONTAINER_REGISTRY_FIELDS)
+
+    # Given an empty string to password
+    variables = {
+        "hostname": "cr.example.com",
+        "props": {
+            "password": "",
+        },
+    }
+
+    # Then password is set to empty string
+    response = await client.execute_async(query, variables=variables, context_value=context)
+    container_registry = response["data"]["modify_container_registry"]["container_registry"]
+    assert container_registry["hostname"] == "cr.example.com"
+    assert container_registry["config"]["url"] == "http://cr.example.com"
+    assert container_registry["config"]["type"] == "harbor2"
+    assert container_registry["config"]["project"] == ["default", "example"]
+    assert container_registry["config"]["username"] == "username2"
+    assert container_registry["config"]["password"] == "*****"
```

Comment:
Also does this test case reveal that the password is actually set as an empty string? I think we need to directly access the etcd to validate the fact.